### PR TITLE
Bump version to 2.2.0

### PR DESCRIPTION
**Pull request by getzze**
Next steps:
- merge to `master` (I would prefer to use `main` but only @Diaoul can change that, so let's use `master` for this release).
- tag the new version
- make a release on github and pypi (any idea how?)

There is only one breaking change that is the default encoding is now 'utf-8' instead of the original encoding, so I think it's fine not to bump to version 3.0

### CODE REVIEW

**Comment by Diaoul** [comment]
> merge to `master` (I would prefer to use `main` but only @Diaoul can change that, so let's use `master` for this release).


Let me know where and when I should do this :+1:

**Comment by getzze** [comment]
> > merge to `master` (I would prefer to use `main` but only @Diaoul can change that, so let's use `master` for this release).
> 
> Let me know where and when I should do this 👍

I just pushed a `main` branch to this repo, you have to make it the default branch in Settings - General - Default Branch. Thanks!

**Comment by getzze** [comment]
I just saw you updated `enzyme`, I'll update the version for the dependency then.

**Comment by Diaoul** [comment]
Yes, please run some tests in case I broke something :laughing:

**Comment by Diaoul** [comment]
@getzze I did something which seems more the way to go
* Changed default branch to develop
* Renamed develop to main

:point_right: this way the PRs are all switched against `main`

**Comment by getzze** [comment]
Ah cool, that's even better!

Tests are passing with the last `enzyme` version 🎉

**Comment by getzze** [comment]
I saw you added a `publish` github action to `enzyme`, if we add it to `subliminal` it will automatically publish the new tagged version to Pypi?

File: subliminal/cli.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def plural(quantity: int, name: str, *, bold: bool = True, **kwargs: Any) -> str
 )
 @click.option(
     '-c',
-    '--config',
+    '--config-file',
```
Comment (ptrcnull) [inline subliminal/cli.py:184]: any reason for the rename? imho `--config` fits better

File: HISTORY.rst
Comment (ptrcnull) [inline HISTORY.rst:1]: would be neat to also include all dependency changes for downstream packagers

Comment (getzze) [reply HISTORY.rst:1]: good point, done!

File: subliminal/cli.py (hunk 1 above)
Comment (getzze) [reply subliminal/cli.py:184]: I think it's more accurate because it expects a file path. Why do you prefer `--config` ?

Comment (ptrcnull) [reply subliminal/cli.py:184]: it feels somewhat more consistent with most other software that i've used, but i guess it's just a personal preference

File: Dockerfile
Comment (getzze) [inline Dockerfile:1]: @DerouineauNicolas I bumped the Docker python version to 3.12, can you check that it's still working please?

**Comment by ptrcnull** [comment]
one little nitpick that i didn't catch earlier: do we have to print `Not using any configuration file.` every single time? wouldn't it be better as a debug print?

File: subliminal/cli.py (hunk 1 above)
Comment (getzze) [reply subliminal/cli.py:184]: At the time, I checked other softwares and I had the feeling that `--config-file` was more used :laughing:
Using `--config` makes it more general, so we don't have to change the name if at some point we want to allow `KEY=VALUE` for instance. `ruff` is like that.
`mypy` uses `--config-file`.

Comment (getzze) [reply subliminal/cli.py:184]: I'll revert then :+1:

**Comment by getzze** [comment]
Yes, I don't like it either.
The problem is that when the config file is parsed, the logger is not set yet therefore the message is "printed".
Another problem I saw is that the two options `--debug` and `--verbose` are redundant.

**Comment by ptrcnull** [comment]
i'd say redundant options are alright, synonymous flags just make it easier for users to get it first try :P

File: Dockerfile
Comment (ptrcnull) [reply Dockerfile:1]: i can also confirm that the Docker container works for me

**Comment by Diaoul** [comment]
> it will automatically publish the new tagged version to Pypi?

Yes, not tagged though, released and published on GitHub
1. That usually start with a tag
2. Then you go do make a new release based on that tag (draft or otherwise), you can decide if beta or not
3. Only when the release is "published" (i.e. not draft anymore) the workflow will trigger and publish on PyPI :+1: 

Automation all the way :robot:

**Comment by getzze** [comment]
I added the `publish.yaml` file and a `.readthedocs.yaml` file also. I will draft a release!

**Comment by ptrcnull** [comment]
good job!! thanks everyone, really happy to see a new release ^^

**Comment by getzze** [comment]
I managed to make the release! Thank you all!
I changed the github action to automatically make a Github release and a Pypi release when a new tag is added.
The documentation in readthedocs was not updated though. If found this [gha](https://github.com/peaceiris/actions-gh-pages), but I am not sure it would work.

**Pull request merged**